### PR TITLE
some fixes for linux guests

### PR DESCRIPTION
**Pull request by Bondey**
Some fixes that makes linux guests work "out of the box"

### CODE REVIEW

**Comment by doomedraven** [comment]
See linux pr ;) there more stuff fixed/improved

**Comment by Bondey** [comment]
linux pr??

**Comment by doomedraven** [comment]
https://github.com/cuckoosandbox/cuckoo/pull/1545

**Comment by doomedraven** [comment]
btw you have some nice improvements, which is missed in my PR, so thanks ++ :)

**Comment by Bondey** [comment]
Ty :3, im working on some nice things about ELFs, will show them soon ;)

**Comment by doomedraven** [comment]
great :) lets boost linux cuckoo

**Comment by doomedraven** [comment]
btw i will backport your changes to my fork which contains this pr https://github.com/cuckoosandbox/cuckoo/pull/1545 so i can always deploy it with pip install :) and there one guy from cuckoo who already started review of my pr to merge that

File: cuckoo/core/guest.py [hunk 1]
```diff
@@ -459,22 +462,42 @@ def start_analysis(self, options, monitor):
...
+                        self.environ["TEMP"], options["file_name"]
+                    ),
+                }
+                files = {
+                    "file": ("sample.bin", open(options["target"], "rb")),
+                }
+            else:
+                data = {
+                    "filepath": os.path.join(
+                        "/tmp", options["file_name"]
```
Comment (doomedraven) [inline cuckoo/core/guest.py]: i would suggest change it to `self.environ["TEMP"]`

**Comment by jbremer** [comment]
Yes, this is on @daanfs' TODO list :-)

**Comment by jbremer** [comment]
@daanfs has all of this been implemented as part of your work (i.e., can this PR be closed)?

**Comment by daanfs** [comment]
@jbremer: Yes, with the addition of https://github.com/daanfs/cuckoo/commit/f854057e6103e5ef4f1e329002f8ae31e9c31108 to the recent merges from master this has all been implemented and can be closed.

**Comment by daanfs** [comment]
@jbremer: Yes, with the addition of https://github.com/daanfs/cuckoo/commit/80884581c881363f36acfba90f90ee063088e4b0 to the recent merges from master this has all been implemented and can be closed.

**Comment by jbremer** [comment]
@daanfs excellent! @Bondey thanks again for your contributions!

**Comment by Bondey** [comment]
Np, would like to have time to do more, its an awesome project!

**Pull request closed without merging**